lib/node-loader-core.js

```diff
@@ -89,11 +114,11 @@ function getConfigPromise() {
           const config = ns.default;
           return processConfig(config);
         },
-        (err) => {
+        (error) => {
           console.warn(
             `Could not read node-loader.config.js file at ${configPath}, continuing without node loader config`
           );
-          console.error(err);
+          // Swallow error and return an empty implementation
```

Comment:
I think this is probably a good change, as the stacktrace can be disorienting to see. One problem, though, is that now the user won't know if the problem is in the file path or in the syntax / instantiation of the config file. I think ideally we'd differentiate between those two errors and surface them to the user in our console warning. Do you know of a way to do that without logging the whole error object?

Reply:
Excellent point about needing to differentiate between error scenarios. The absence of a config, and the fallback to an empty implementation warrants a warning, but problems with the path and config syntax definitely warrant a helpful user message as to the cause of the problem. 

To achieve that, exceptions would need to be explicitly caught and re-thrown closer in the code flow to the where those problems can occur, and the re-thrown errors would be tagged in some way as a certain type (message, code, custom Error). In the case of exceptions processing the config, at least for syntax errors, I can't think of a better way to surface that than a stack trace.

If you'd like I can try to enhance the error handling here?

Reply:
Showing the stacktrace is fine, and probably necessary for syntax errors. Perhaps we could use `fs.stat()` inside of the `catch`, to see if the file exists. If the file exists, print the whole stacktrace. Otherwise print a console warning that says that the file path doesn't exist.